### ISSUE TITLE
Filename input box in save game menu doesn't have initial focus

### ISSUE THREAD

**Issue by attet** [widelands/widelands#5390]
**Describe the bug**
When opening the menu for saving a game, the filename input box doesn't truly have focus.
The caret is blinking as if the box is focused, yet nothing happens when characters are typed. I need to click on the box before it actually starts accepting input.

Note that this behaviour does not occur if the save game menu is instead opened using keyboard shortcut (Ctrl+ S); in that case the field is immediately accepting input as expected.

**To Reproduce**
1. Start/load any game
2. Click 'main menu'
3. Click 'save game'
4. The filename text box is blinking as if ready for input
5. Type a character. Nothing happens.

**Version:**
 - OS: Windows 11
 - Widelands Version: recent master

**Comment by akupar** [comment]
It looks like this is because the main menu dropdown disables SDL text input when it is closed on line [554](https://github.com/widelands/widelands/blob/b3541945910bd9bb4e1678433b8c3ee37ac441e9/src/ui_basic/dropdown.cc#L554) of dropdown.cc. If you remove that line, the issue with save game dialog (and the debug console) is gone, but I think removing it would cause some other issues (see #5281).

Edit: Also this #3351.

**Comment by tothxa** [comment]
I think an important aspect of this is that the caret shouldn't be blinking (best it should be hidden) when a text input field doesn't have focus. Or should this be a separate issue?

**Comment by akupar** [comment]
> I think an important aspect of this is that the caret shouldn't be blinking (best it should be hidden) when a text input field doesn't have focus.

It actually does have focus. It just doesn't get input, because the SDL text input is disabled.
